runtime/tests/tests.rs

```diff
@@ -222,35 +263,318 @@ fn generic_asset_transfer_works_with_fee_exchange() {
 
 #[test]
 fn contract_fails() {
-	// Contract itself fails
+	ExtBuilder::default()
+		.initial_balance(1_000_000 * TransactionBaseFee::get())
+		.gas_price(1)
+		.build()
+		.execute_with(|| {
+			let contract_address = setup_contract(mock::contracts::CONTRACT_WITH_TRAP, alice());
+
+			// Call the newly instantiated contract. The contract is expected to dispatch a call
+			// and then trap.
+			let contract_call = Call::Contracts(pallet_contracts::Call::call(
+				contract_address, // newly created contract address
+				0,                // transfer value in
+				1_000_000,        // gas limit
+				vec![],
+			));
+			let contract_call_extrinsic = sign(CheckedExtrinsic {
+				signed: Some((bob(), signed_extra(2, 0, None, None))),
+				function: contract_call,
+			});
+
+			assert!(Executive::apply_extrinsic(contract_call_extrinsic).is_err());
+		});
+}
+
+#[test]
+fn contract_call_fails_with_insufficient_gas_without_fee_exchange() {
+	ExtBuilder::default()
+		.initial_balance(10_000 * TransactionBaseFee::get())
+		.gas_price(1)
+		.build()
+		.execute_with(|| {
+			initialize_block();
+			let xt = sign(CheckedExtrinsic {
+				signed: Some((alice(), signed_extra(0, 0, None, None))),
+				function: Call::Contracts(pallet_contracts::Call::call::<Runtime>(bob(), 10, 1000, vec![])),
```

Comment:
This test fails because we don't provide a high enough gas limit
The purpose of this test is to check that our paying for gas fails because the user doesnt have enough fee assets in thier account - we should be setting `gas_limit` high and `initial_balance` low

Reply:
Okay we trust plug for handling `gas_limit` too low case